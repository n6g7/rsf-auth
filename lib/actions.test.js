import {
  login,
  loginSuccess,
  loginFailure,
  logout,
  logoutSuccess,
  logoutFailure,
  syncUser
} from './actions'

describe('actions', () => {
  describe('login()', () => {
    it('works', () => {
      expect(login()).toMatchSnapshot()
    })
  })

  describe('loginSuccess(credential)', () => {
    it('works', () => {
      const credential = 'qsdk'
      expect(loginSuccess(credential)).toMatchSnapshot()
    })
  })

  describe('loginFailure(error)', () => {
    it('works', () => {
      const error = 'qsdk'
      expect(loginFailure(error)).toMatchSnapshot()
    })
  })

  describe('logout()', () => {
    it('works', () => {
      expect(logout()).toMatchSnapshot()
    })
  })

  describe('logoutSuccess()', () => {
    it('works', () => {
      expect(logoutSuccess()).toMatchSnapshot()
    })
  })

  describe('logoutFailure(error)', () => {
    it('works', () => {
      const error = 'qsdk'
      expect(logoutFailure(error)).toMatchSnapshot()
    })
  })

  describe('syncUser(user)', () => {
    it('works', () => {
      const user = 'qsdk'
      expect(syncUser(user)).toMatchSnapshot()
    })
  })
})
