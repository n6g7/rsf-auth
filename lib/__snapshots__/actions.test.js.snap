// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions login() works 1`] = `
Object {
  "type": "@RSFAUTH_LOGIN.REQUEST",
}
`;

exports[`actions loginFailure(error) works 1`] = `
Object {
  "error": "qsdk",
  "type": "@RSFAUTH_LOGIN.FAILURE",
}
`;

exports[`actions loginSuccess(credential) works 1`] = `
Object {
  "credential": "qsdk",
  "type": "@RSFAUTH_LOGIN.SUCCESS",
}
`;

exports[`actions logout() works 1`] = `
Object {
  "type": "@RSFAUTH_LOGOUT.REQUEST",
}
`;

exports[`actions logoutFailure(error) works 1`] = `
Object {
  "error": "qsdk",
  "type": "@RSFAUTH_LOGOUT.FAILURE",
}
`;

exports[`actions logoutSuccess() works 1`] = `
Object {
  "type": "@RSFAUTH_LOGOUT.SUCCESS",
}
`;

exports[`actions syncUser(user) works 1`] = `
Object {
  "type": "@RSFAUTH_SYNC_USER",
  "user": "qsdk",
}
`;
